_posts/2015-10-19-introduction_to_clojurescript.md

```diff
@@ -0,0 +1,269 @@
+
+user=> 
+```
+
+We're now in a ClojureScript Repl. This will allow us quickly execute ClojureScript and view the result.
+
+
+## ClojureScript Syntax
+ClojureScript is a functional language. This means it has functions and limited additional language constructs. It is a Lisp style language.
+
+### Primitives
+
+* Number
+
+    ```
+    user=> 1.23
+    1.23
+    ```
+
+
+* String
+
+    ```
+    user=> "foo"
+    "foo"
+    ```
+
+
+* Vector (array)
+    
+    ```
+    user=> [:bar 3.14 "hello"]
+    [:bar 3.14 "hello"]
+    ```
+
+
+* Map (associative arrays)
+
+    ```
+    user=> {:msg "hello" :pi 3.14 :primes [2 3 5 7 11 13]}
+    {:msg "hello", :pi 3.14, :primes [2 3 5 7 11 13]}
+    ```
+
+* Keyword (used to access Maps)
+
+    ```
+    user=> :foo
+    :foo
+    ```
+
+
+* Set (distinct array)
+    
+    ```
+    user=> #{:bar 3.14 "hello"}
+    #{"hello" 3.14 :bar}
+    ```
+
+### Functions all the way down
+Functions are the building blocks of ClojureScript. To define a function, you even call a function `defn`.
+
+```
+user=> (defn myfunction [argument1] argment1)
+```
+Here we've defined a function named `myfunction`. It takes one argument `argument1` and returns it. It's not a very useful function but it's a good example of syntax. This is equivalent to the following Javascript:
+
+```javascript
+function myfunction(argument1){
+    return argument1;
+}
+```
+
+Functions are invoked by wrapping their name and arguments with parentheses:
+
+```
+user=> (myfunction "hello world")
+"hello world"
+```
+
+In non-functional programming languages there are special "Operators" or keywords. 
+In Javascript, some commonly used operators are are `+ - == if`. In ClojureScript and other Lisp languages, there are no special operators. These are just regular functions. 
+
+If statements are functions
+
+```
+user=> (if true "do true stuff here" "do false stuff here")
+"do true stuff here"
+```
+
+Math operators are functions
+
+```
+user=> (+ 2 3)
+5
+user=> (* 2 3)
+6
+```
+
+For more great examples of [Javascript to ClojureScript synonyms checkout this site](https://kanaka.github.io/clojurescript/web/synonym.html)
+
+## Creating a Node.js - ClojureScript project
+Starting a ClojureScript project is simple. Leningen offers project templates that will get you up and running with a boilerplate project.
+Templates are a great resource to play around with and see other uses and configurations for ClojureScript projects. [Clojars.org has a collection of templates](https://clojars.org/search?q=template) and others can be found searching the web. For our project we'll be using a [Nodejs Figwheel project template](https://github.com/malyn/figwheel-node-template). At your terminal, run:
+
+```
+$ lein new figwheel-node hello-world
+$ cd hello-world
+$ npm install
+```
+
+This will create a new ClojureScript project in the directory `./hello-world`. Change to that directory and install NPM dependencies.
```

Comment:
probably a few opportunities to watch out for passive voice and shorten sentences: "this creates a new" vs "this will create". second sentence can probably be removed here, or made more specific as to what you're referring to: "Running `npm install` sets up the libraries the closurescript project needs to run". Right now I think it's just a duplicate sentence